pkg/aws/actuator/actuator.go

```diff
@@ -326,11 +330,18 @@ func (a *AWSActuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest)
 		logger.Debug("credentials already up to date")
 		return nil
 	}
-	stsDetected := utils.IsTimedTokenCluster(a.Client, logger)
+	stsDetected, _ := utils.IsTimedTokenCluster(a.Client, logger)
 	if stsDetected {
-		logger.Info("actuator detected STS enabled cluster making secret")
-		if cr.Spec.CloudTokenString != "" {
-			err = a.createSecret(cr.Spec.CloudTokenString, cr.Spec.CloudTokenPath, cr.Spec.SecretRef.Name, cr.Spec.SecretRef.Namespace, logger)
+		if a.Codec == nil {
+			return fmt.Errorf("invalid codec, nil value")
+		}
+		logger.Info("actuator detected STS enabled cluster, enabling STS secret brokering for CredentialsRequests providing an IAM Role ARN")
+		awsSTSIAMRoleARN, err := awsSTSIAMRoleARN(a.Codec, cr)
+		if err != nil {
+			return err
+		}
+		if awsSTSIAMRoleARN != "" {
+			err = a.createSecret(awsSTSIAMRoleARN, cr.Spec.CloudTokenPath, cr.Spec.SecretRef.Name, cr.Spec.SecretRef.Namespace, logger)
```

Comment:
`CloudTokenPath` is an optional field so we should use "/var/run/secrets/kubernetes.io/serviceaccount/token" as the path when it is not found.

Reply:
got done.